ParquetClassLibrary/Characters/Character.cs

```diff
@@ -40,29 +64,35 @@ public abstract class Character : Being
         /// <param name="in_primaryBehavior">The rules that govern how this <see cref="Character"/> acts.  Cannot be null.</param>
         /// <param name="in_avoids">Any parquets this <see cref="Character"/> avoids.</param>
         /// <param name="in_seeks">Any parquets this <see cref="Character"/> seeks.</param>
-        /// <param name="in_quests">Any quests this <see cref="Character"/> has to offer.</param>
-        /// <param name="in_dialogue">All dialogue this <see cref="Character"/> may say.</param>
-        /// <param name="in_inventory">Any items this <see cref="Character"/> owns.</param>
         /// <param name="in_pronoun">How to refer to this <see cref="Character"/>.</param>
-        protected Character(Range<EntityID> in_bounds, EntityID in_id, string in_name, Biome in_nativeBiome,
+        /// <param name="in_storyCharacterID">A means of identifying this <see cref="Character"/> across multiple shipped game titles.</param>
+        /// <param name="in_startingQuests">Any quests this <see cref="Character"/> has to offer or has undertaken.</param>
+        /// <param name="in_dialogue">All dialogue this <see cref="Character"/> may say.</param>
+        /// <param name="in_startingInventory">Any items this <see cref="Character"/> owns at the outset.</param>
+        protected Character(Range<EntityID> in_bounds, EntityID in_id,
+                            string in_personalName, string in_familyName, Biome in_nativeBiome,
                             Behavior in_primaryBehavior, List<EntityID> in_avoids = null,
-                            List<EntityID> in_seeks = null, List<EntityID> in_quests = null,
-                            List<string> in_dialogue = null, List<EntityID> in_inventory = null,
-                            string in_pronoun = DefaultPronoun)
-            : base(in_bounds, in_id, in_name, in_nativeBiome, in_primaryBehavior, in_avoids, in_seeks)
+                            List<EntityID> in_seeks = null, string in_pronoun = DefaultPronoun,
+                            string in_storyCharacterID = "", List<EntityID> in_startingQuests = null,
+                            List<string> in_dialogue = null, List<EntityID> in_startingInventory = null)
+            : base(in_bounds, in_id, in_personalName, in_nativeBiome, in_primaryBehavior, in_avoids, in_seeks)
         {
             var nonNullPronoun = string.IsNullOrEmpty(in_pronoun) ? DefaultPronoun : in_pronoun;
             var nonNullQuests = in_quests ?? Enumerable.Empty<EntityID>();
             var nonNullInventory = in_inventory ?? Enumerable.Empty<EntityID>();
 
             Precondition.AreInRange(nonNullQuests, All.QuestIDs, nameof(in_quests));
             Precondition.AreInRange(nonNullInventory, All.ItemIDs, nameof(in_inventory));
+            Precondition.NotEmpty(in_personalName, nameof(in_personalName));
+            Precondition.NotEmpty(in_familyName, nameof(in_familyName));
```

Comment:
In the design document, there were a lot of characters with empty family names. There's also the weird question of, "what is the cannon ant's family name". Is it ant?

Reply:
Goooooooood point!
So, for the characters whose family names is not given, I was expecting to fill that in later.
But for the Cannon Ant there is the possibility that they genuinely don't have a family name....
I will think on this!